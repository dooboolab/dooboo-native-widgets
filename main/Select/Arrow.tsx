import Svg, { Line } from 'react-native-svg';
import { TextStyle, View } from 'react-native';

import React from 'react';

interface Props {
  customColor?: TextStyle;
}

export default function Arrow(props: Props): React.ReactElement {
  const { customColor = { color: '#2b2b2b' } } = props;
  return (
    <View>
      <Svg height="20" width="20" viewBox="0 0 20 20">
        <Line
          x1={5}
          y1={7}
          x2={10}
          y2={12}
          stroke={customColor.color || '#2b2b2b'}
          strokeWidth="1.5"
          strokeLinecap="round"
          opacity={0.95}
        />
        <Line
          x1={10}
          y1={12}
          x2={15}
          y2={7}
          stroke={customColor.color || '#2b2b2b'}
          strokeWidth="1.5"
          strokeLinecap="round"
          opacity={0.95}
        />
      </Svg>
    </View>
  );
}
