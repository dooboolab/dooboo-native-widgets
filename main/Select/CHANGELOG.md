[0.0.1]
- First release.

[0.0.2]
- Removed theme feature
- Fixed styling & customize feature
