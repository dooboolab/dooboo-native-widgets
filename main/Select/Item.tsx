import { TextStyle, ViewStyle } from 'react-native';

import React from 'react';
import styled from 'styled-components/native';

interface Props {
  testID?: string;
  activeOpacity?: number;
  containerHeight?: number;
  value: string | number;
  onSelectItem?:((param: string | number) => void) | undefined;
  customStyle?: ViewStyle
  customTextStyle?: TextStyle;
  children?: string | number;
}
const ItemWrapper = styled.TouchableOpacity`
  width: 100%;
  padding: 14px 12px;
  justify-content: center;
`;
const ItemText = styled.Text`
  font-size: 12px;
  color: #2b2b2b;
`;
const Item: React.FC<Props> = (props): React.ReactElement => {
  const {
    testID,
    children = '',
    onSelectItem = (value): string | number => value,
    containerHeight,
    activeOpacity = 0.75,
    value,
    customStyle,
    customTextStyle,
  } = props;

  React.useEffect(() => {}, [value]);
  return (
    <ItemWrapper
      testID={testID}
      activeOpacity={activeOpacity}
      style={[{ height: containerHeight }, customStyle]}
      onPress={(): void => onSelectItem(value)}>
      <ItemText style={customTextStyle}>
        {children}
      </ItemText>
    </ItemWrapper>
  );
};

export default Item;
