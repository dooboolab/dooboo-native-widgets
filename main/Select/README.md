# Select

- [![Npm Version](http://img.shields.io/npm/v/@dooboo-ui/native-select.svg?style=flat-square)](https://npmjs.org/package/@dooboo-ui/native-select)
- [![Downloads](http://img.shields.io/npm/dm/@dooboo-ui/native-select.svg?style=flat-square)](https://npmjs.org/package/@dooboo-ui/native-select)

(Image Area: coming soon)

## Props

|                | necessary | types                      | default |
| -------------- | --------- | -------------------------- | ------- |
| testID         |           | string                     |         |
| label          |           | string                     |         |
| items          | ‚úì         | array                      |         |
| placeholder    |           | string                     |         |
| theme          |           | 'underbar', 'box', 'none'  | `none`  |
| rootStyle      |           | ViewStyle | TextStyle      |         |
| itemStyle      |           | ViewStyle | TextStyle      |         |
| disabled       |           | boolean                    | `false` |
| [onFocus](#onFocus)   |           | func                       |         |
| [onBlur](#onBlur)     |           | func                       |         |
| [onSelect](#onSelect) |           | func                       |         |

## description
### onFocus
...onFocus

### onBlur
...onFocus

### onSelect
...onFocus

## Installation

```sh
yarn add @dooboo-ui/native
```

or

```sh
yarn add @dooboo-ui/native-select
```

## Getting started

- Import

  ```javascript
  import { Select } from '@dooboo-ui/native';
  ```

- Usage
  ```jsx
  function Page(props: Props) {
    return (
      <Container>
        <Button
          testID="btn"
          isLoading={false}
          onClick={() => {}}
        >
          üòÄ üòé üëç üíØ
        </Button>
        <Button
          style={{
            marginVertical: 40,
          }}
          isDisabled={true}
          onClick={() => {}}
        >
          This is disabled!!
        </Button>
        <Button
          testID="btnGoogle"
          iconLeft={<Image source={IC_GOOGLE} />}
          isLoading={googleLoading}
          indicatorColor="#023059"
          onClick={() => {
            setGoogleLoading(true);
            const timeout = setTimeout(() => {
              setGoogleLoading(false);
              clearTimeout(timeout);
            }, 2000);
          }}
        >
          GOOGLE SIGN IN
        </Button>
        <Button
          testID="btnFacebook"
          iconLeft={<Image source={IC_FACEBOOK} />}
          indicatorColor="#023059"
          isLoading={facebookLoading}
          style={{
            marginTop: 40,
            backgroundColor: '#ccc',
            borderWidth: 0.5,
            borderRadius: 0,
          }}
          onClick={() => {
            setFacebookLoading(true);
            const timeout = setTimeout(() => {
              setFacebookLoading(false);
              clearTimeout(timeout);
            }, 2000);
          }}
        >
          FACEBOOK SIGN IN
        </Button>
      </Container>
    );
  }
  ```