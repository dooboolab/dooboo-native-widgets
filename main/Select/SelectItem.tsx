import { TextStyle, ViewStyle } from 'react-native';

import React from 'react';
import styled from 'styled-components/native';

export interface SelectItemProps {
  testID?: string;
  value: string;
  onSelectItem?: ((param: string) => void) | undefined;
  style?: ViewStyle;
  textStyle?: TextStyle;
  children?: string;
}

const ItemWrapper = styled.TouchableOpacity.attrs({ activeOpacity: 0.9 })`
  width: 100%;
  padding: 14px 12px;
  justify-content: center;
`;

const ItemText = styled.Text`
  font-size: 12px;
  color: #2b2b2b;
`;

const SelectItem: React.FC<SelectItemProps> = (props): React.ReactElement => {
  const {
    testID,
    children = '',
    onSelectItem = (value): string => value,
    value,
    style,
    textStyle,
  } = props;

  return (
    <ItemWrapper
      testID={testID}
      style={style}
      onPress={(): void => onSelectItem(value)}>
      <ItemText style={textStyle}>{children}</ItemText>
    </ItemWrapper>
  );
};

export default SelectItem;
