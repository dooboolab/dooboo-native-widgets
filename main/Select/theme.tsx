
const COLOR: {
  [key: string]: string;
} = {
  WHITE: '#ffffff',
  DODGERBLUE: '#3a8bff',
  VERYLIGHTGRAY: '#cccccc',
  LIGHTGRAY: '#c8c8c8',
  BLUE: '#0000ff',
  LIGHTBLUE: '#c9e9fc',
  STRONGBLUE: '#069ccd',
  GRAY3: '#080808',
  GRAY7: '#121212',
  GRAY59: '#969696',
};

interface ThemeType {
  backgroundColor: string;
  borderColor: string;
  fontColor: string;
}
interface StatefulThemeType extends ThemeType {
  INVERTED: ThemeType;
  DISABLED: ThemeType;
}

export const THEME: {
  LIGHT: StatefulThemeType;
  DARK: StatefulThemeType;
} = {
  LIGHT: {
    backgroundColor: COLOR.WHITE,
    borderColor: COLOR.LIGHTGRAY,
    fontColor: COLOR.GRAY7,
    INVERTED: {
      backgroundColor: COLOR.WHITE,
      borderColor: COLOR.LIGHTGRAY,
      fontColor: COLOR.GRAY7,
    },
    DISABLED: {
      backgroundColor: COLOR.VERYLIGHTGRAY,
      borderColor: COLOR.LIGHTGRAY,
      fontColor: COLOR.GRAY59,
    },
  },
  DARK: {
    backgroundColor: COLOR.GRAY7,
    borderColor: COLOR.LIGHTGRAY,
    fontColor: COLOR.WHITE,
    INVERTED: {
      backgroundColor: COLOR.GRAY59,
      borderColor: COLOR.LIGHTGRAY,
      fontColor: COLOR.WHITE,
    },
    DISABLED: {
      backgroundColor: COLOR.VERYLIGHTGRAY,
      borderColor: COLOR.LIGHTGRAY,
      fontColor: COLOR.GRAY59,
    },
  },
};
