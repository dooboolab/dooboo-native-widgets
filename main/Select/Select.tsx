import {
  Animated,
  Easing,
  TextStyle,
  ViewStyle,
} from 'react-native';
import React, { Fragment, ReactNode } from 'react';

import Arrow from './Arrow';
import { LoadingIndicator } from '../LoadingIndicator';
import styled from 'styled-components/native';

interface Props {
  testID?: string;
  opened?: boolean;
  loading?: boolean;
  disabled?: boolean;
  showArrow?: boolean;
  bordered?: boolean;
  activeOpacity?: number;
  listHeight?: number;
  placeholder?: string | number;
  value?: string | number;
  defaultValue?: string | number;
  onSelect?: (param: string | number) => void;
  onOpen?: (param: boolean) => void;
  prefixIcon?: ReactNode;
  suffixIcon?: ReactNode;
  customLoader?: ReactNode;
  customStyle?: ViewStyle
  customTextStyle?: TextStyle;
  children?: ReactNode;
}

const Container = styled.View`
  height: 100%;
  width: 100%;
`;

const SelectWrapper = styled.TouchableOpacity<{
  disabled: boolean;
  bordered: boolean;
}>`
  width: 100%;
  height: 100%;
  flex-direction: row;
  align-items: center;
  padding: 14px 12px;
  background-color: #f7f7f7;
  border-radius: ${({ bordered }): string => (bordered ? '5px' : '0')};
  border-color: ${({ disabled }): string => disabled ? '#c8c8c8' : 'transparent'};
  box-shadow: 0px 2px 4px rgba(212, 210, 212, 0.8);
  opacity: 0.9;
`;

const SelectChildWrapper = styled.View<{
  visible: boolean;
  bordered: boolean;
}>`
  border-color: transparent;
  background-color: #ffffff;
  border-bottom-left-radius: ${({ bordered }): string =>
    bordered ? '5px' : '0'};
  border-bottom-right-radius: ${({ bordered }): string =>
    bordered ? '5px' : '0'};
  border-width: ${({ visible }): string => (visible ? '1px' : '0px')};
  box-shadow: 0px 2px 4px rgba(212, 210, 212, 0.8);
`;

const IconView = styled.View`
  width: 20px;
  height: 20px;
  justify-content: center;
  align-items: center;
`;

const StyledText = styled.Text<{disabled: boolean}>`
  align-self: center;
  font-size: 12px;
  color: ${({ disabled }): string => disabled ? '#969696' : '#2b2b2b'};
`;

const Select : React.FC<Props> = (props): React.ReactElement => {
  const {
    testID,
    opened = false,
    loading = false,
    disabled = false,
    showArrow = true,
    bordered = true,
    activeOpacity = 0.75,
    listHeight,
    placeholder,
    value,
    defaultValue,
    onSelect = (value): string | number => value,
    onOpen = (opened): boolean => opened,
    prefixIcon,
    suffixIcon,
    customLoader,
    customStyle,
    customTextStyle,
    children,
  } = props;

  const [containerHeight, setContainerHeight] = React.useState(10);
  const rotateAnimValue = React.useRef(new Animated.Value(0)).current;
  const slideAnimValue = React.useRef(new Animated.Value(0)).current;

  React.useEffect(() => {
    const rotateValue = opened ? 1 : 0;
    Animated.timing(rotateAnimValue, {
      toValue: rotateValue,
      duration: 120,
      easing: Easing.linear,
      useNativeDriver: true,
    }).start();

    const slideValue = !disabled && opened ? 1 : 0;
    Animated.timing(slideAnimValue, {
      toValue: slideValue,
      duration: 120,
      easing: Easing.linear,
      useNativeDriver: false,
    }).start();
  }, [opened, disabled]);

  React.useEffect(() => {
  }, [value, bordered, loading]);

  return (
    <Container testID={testID}>
      <SelectWrapper
        testID={'SELECT_WRAPPER'}
        onLayout={(e): void => setContainerHeight(e.nativeEvent.layout.height)}
        activeOpacity={0.95}
        disabled={loading ? true : disabled}
        bordered={bordered}
        style={[
          {
            borderBottomLeftRadius: opened ? 0 : bordered ? 5 : 0,
            borderBottomRightRadius: opened ? 0 : bordered ? 5 : 0,
          },
          customStyle,
        ]}
        onPress={(): void => onOpen(!opened)}>
        <Fragment>
          {prefixIcon && <IconView>{prefixIcon}</IconView>}
          {
            <StyledText
              disabled={loading}
              style={[customTextStyle]}>
              {defaultValue || value || placeholder}
            </StyledText>
          }
        </Fragment>
        <Animated.View
          testID={'SELECT_SUFFIX'}
          style={[
            {
              position: 'absolute',
              right: 10,
              transform: [
                {
                  rotate: rotateAnimValue.interpolate({
                    inputRange: [0, 1],
                    outputRange: ['0deg', '180deg'],
                  }),
                },
              ],
            },
            customStyle,
          ]}>
          <IconView>
            {(loading && (customLoader || <LoadingIndicator size="small" />)) ||
              suffixIcon ||
              (showArrow && (
                <Arrow
                  customColor={customTextStyle}
                />
              ))}
          </IconView>
        </Animated.View>
      </SelectWrapper>
      <SelectChildWrapper
        visible={opened}
        bordered={bordered}
        style={[customStyle]}>
        <Animated.ScrollView
          testID={'SELECT_CHILD_SCROLLVIEW'}
          keyboardDismissMode={'on-drag'}
          style={[
            {
              height: slideAnimValue.interpolate({
                inputRange: [0, 1],
                outputRange: [
                  0,
                  listHeight ||
                    containerHeight * React.Children.count(children) + 1,
                ],
              }),
            },
            customStyle,
          ]}>
          <Fragment>
            {React.Children.map(children, (child) => {
              if (React.isValidElement(child)) {
                return React.cloneElement(child, {
                  onSelectItem: (value) => {
                    onOpen(!opened);
                    return onSelect(value);
                  },
                  activeOpacity,
                  containerHeight,
                });
              }
              return child;
            })}
          </Fragment>
        </Animated.ScrollView>
      </SelectChildWrapper>
    </Container>
  );
};

export default Select;
