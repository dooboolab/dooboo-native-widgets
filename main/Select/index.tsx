import Item from './Item';
import Select from './Select';

export { Select, Item };
