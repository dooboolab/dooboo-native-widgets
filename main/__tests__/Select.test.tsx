import React, { ReactElement } from 'react';
import {
  RenderResult,
  cleanup,
  render,
} from '@testing-library/react-native';
import { Select, SelectItem } from '../Select';

let props: any;
let component: ReactElement;
let testingLib: RenderResult;

const createTestProps = (
  obj?: Record<string, unknown>,
): Record<string, unknown> => ({
  ...obj,
});

describe('[Select] render test', () => {
  const placeholderText = 'Testing-placeholder';
  it('should render without crashing', () => {
    props = createTestProps();
    component = <Select {...props} placeholder={placeholderText} />;
    testingLib = render(component);

    expect(testingLib.baseElement).toMatchSnapshot();
    expect(testingLib.baseElement).toBeTruthy();
  });
});

describe('[SelectItem] render test', () => {
  const itemValue = 'Item-1';
  it('should render without crashing', () => {
    props = createTestProps();
    component = <Select {...props} placeholder={itemValue} />;
    testingLib = render(component);

    expect(testingLib.baseElement).toMatchSnapshot();
    expect(testingLib.baseElement).toBeTruthy();
  });
});

describe('[Select, Item] event test', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    props = createTestProps({
      disabled: false,
      opened: true,
      showArrow: true,
    });
    component = (
      <Select {...props}>
        <SelectItem value="Item-1">{'Item-1'}</SelectItem>
        <SelectItem value="Item-2">{'Item-2'}</SelectItem>
      </Select>
    );
    testingLib = render(component);
    jest.runAllTimers();
  });

  it('when Disabled is false and Opened is true should operate dropdown animation', () => {
    const selectChild = testingLib.getByTestId('SELECT_CHILD_SCROLLVIEW');

    expect(selectChild.props.style.height).toBeGreaterThan(0);
  });

  it('when Disabled is false and Opened is true should operate arrow rotate animation', () => {
    const selectSuffix = testingLib.getByTestId('SELECT_SUFFIX');

    expect(selectSuffix.props.style.transform[0].rotate).toBe('180deg');
  });

  afterAll((done) => {
    cleanup();
    done();
  });
});
