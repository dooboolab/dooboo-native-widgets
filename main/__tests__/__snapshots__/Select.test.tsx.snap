// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Item] render test should render without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <
        activeOpacity={0.95}
        bordered={true}
        disabled={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f7f7f7",
              "borderColor": "transparent",
              "borderRadius": 5,
              "flexDirection": "row",
              "height": "100%",
              "opacity": 0.9,
              "paddingBottom": 14,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 14,
              "shadowColor": "rgba(212, 210, 212, 0.8)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": "100%",
            },
            Object {
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
            },
            undefined,
          ]
        }
        testID="SELECT_WRAPPER"
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#2b2b2b",
                "fontSize": 12,
              },
              undefined,
            ]
          }
        >
          Item-1
        </Text>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 10,
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
          testID="SELECT_SUFFIX"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                },
              ]
            }
          >
            <View>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                focusable={false}
                height="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={20}
                vbWidth={20}
                width="20"
              >
                <RNSVGGroup>
                  <RNSVGLine
                    opacity={0.95}
                    propList={
                      Array [
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                      ]
                    }
                    stroke={4281019179}
                    strokeLinecap={1}
                    strokeWidth="1.5"
                    x1={5}
                    x2={10}
                    y1={7}
                    y2={12}
                  />
                  <RNSVGLine
                    opacity={0.95}
                    propList={
                      Array [
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                      ]
                    }
                    stroke={4281019179}
                    strokeLinecap={1}
                    strokeWidth="1.5"
                    x1={10}
                    x2={15}
                    y1={12}
                    y2={7}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </>
      <View
        bordered={true}
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "transparent",
              "borderWidth": 0,
              "shadowColor": "rgba(212, 210, 212, 0.8)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
            },
            undefined,
          ]
        }
        visible={false}
      >
        <ScrollView
          keyboardDismissMode="on-drag"
          scrollEventThrottle={0.0001}
          style={
            Object {
              "height": 0,
            }
          }
          testID="SELECT_CHILD_SCROLLVIEW"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`[Select] render test should render without crashing 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    >
      <
        activeOpacity={0.95}
        bordered={true}
        disabled={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#f7f7f7",
              "borderColor": "transparent",
              "borderRadius": 5,
              "flexDirection": "row",
              "height": "100%",
              "opacity": 0.9,
              "paddingBottom": 14,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 14,
              "shadowColor": "rgba(212, 210, 212, 0.8)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
              "width": "100%",
            },
            Object {
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
            },
            undefined,
          ]
        }
        testID="SELECT_WRAPPER"
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#2b2b2b",
                "fontSize": 12,
              },
              undefined,
            ]
          }
        >
          Testing-placeholder
        </Text>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 10,
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
          testID="SELECT_SUFFIX"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "height": 20,
                  "justifyContent": "center",
                  "width": 20,
                },
              ]
            }
          >
            <View>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="20"
                bbWidth="20"
                focusable={false}
                height="20"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "flex": 0,
                      "height": 20,
                      "width": 20,
                    },
                  ]
                }
                vbHeight={20}
                vbWidth={20}
                width="20"
              >
                <RNSVGGroup>
                  <RNSVGLine
                    opacity={0.95}
                    propList={
                      Array [
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                      ]
                    }
                    stroke={4281019179}
                    strokeLinecap={1}
                    strokeWidth="1.5"
                    x1={5}
                    x2={10}
                    y1={7}
                    y2={12}
                  />
                  <RNSVGLine
                    opacity={0.95}
                    propList={
                      Array [
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                      ]
                    }
                    stroke={4281019179}
                    strokeLinecap={1}
                    strokeWidth="1.5"
                    x1={10}
                    x2={15}
                    y1={12}
                    y2={7}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </>
      <View
        bordered={true}
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "transparent",
              "borderWidth": 0,
              "shadowColor": "rgba(212, 210, 212, 0.8)",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 4,
            },
            undefined,
          ]
        }
        visible={false}
      >
        <ScrollView
          keyboardDismissMode="on-drag"
          scrollEventThrottle={0.0001}
          style={
            Object {
              "height": 0,
            }
          }
          testID="SELECT_CHILD_SCROLLVIEW"
        />
      </View>
    </View>
  </View>
</View>
`;
